inference-engine/src/vpu/graph_transformer/src/middleend/passes/gemm_transpose.cpp

```diff
@@ -56,23 +56,36 @@ void PassImpl::run(const Model& model) {
 
         const auto inputDimsA = inputA->desc().dims();
 
-        const auto K = inputDimsA[Dim::H];
-        const auto M = inputDimsA[Dim::W];
-        const auto batch1 = inputDimsA[Dim::N];
-        const auto batch2 = inputDimsA[Dim::C];
+        VPU_THROW_UNLESS(inputDimsA.size() >= 2,
```

Comment:
Should we also check size of `inputDimsB`?

Reply:
In general yes. I think it's not appropriate to do it there, since we are working only with the first input there, but I've added this check for all inputs into GEMM parser.